source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1027,48 +1027,16 @@ void ShaderGraph::optimize(GenContext& context)
         }
         else if (node->hasClassification(ShaderNode::Classification::DOT))
         {
-            // Dot nodes without modifiers can be elided by moving their connection downstream.
+            // Filename dot nodes must be elided so they do not create extra samplers.
             ShaderInput* in = node->getInput("in");
-            if (in->getChannels().empty())
+            if (in->getChannels().empty() && in->getType() == Type::FILENAME)
```

Comment:
Still solves the issue with extra samplers, but preserves the distinction between "dot" nodes and "constant" nodes.